src/main/java/internetStore/servlets/PhoneServlet.java

```diff
@@ -1,2 +1,94 @@
-package internetStore.servlets;public class PhoneServlet {
+package internetStore.servlets;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import internetStore.dao.PhoneDao;
+import internetStore.models.Phone;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.annotation.WebServlet;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@WebServlet("/phones/*")
+public class PhoneServlet extends HttpServlet {
+    private final PhoneDao phoneDao = PhoneDao.getInstance();
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        String contextPath = req.getPathInfo();
+        resp.setContentType("application/json");
+
+        if (contextPath == null || contextPath.equals("/")) {
+            List<Phone> allPhones = phoneDao.getAllPhone();
+            String json = objectMapper.writeValueAsString(allPhones);
+            resp.getWriter().println(json);
+        } else {
+            String[] pathParts = contextPath.split("/");
+            if (pathParts.length == 2) {
+                try {
+                    int phoneId = Integer.parseInt(pathParts[1]);
+                    Phone phone = phoneDao.getById(phoneId);
+                    if (phone != null) {
+                        String json = objectMapper.writeValueAsString(phone);
+                        resp.getWriter().println(json);
+                    } else {
+                        resp.setStatus(HttpServletResponse.SC_NOT_FOUND);
+                        resp.getWriter().println("Phone not found");
+                    }
+                } catch (NumberFormatException e) {
+                    resp.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+                    resp.getWriter().println("Invalid phone ID format");
```

Comment:
This all logic that retrieve ID from context path we can move to another service (or at least a method, but, if we are going to have other servlets that needs to retrieve id? So, create some util class that will be responsible for that).

And use it in delete method, because in doDelete you need ID as well